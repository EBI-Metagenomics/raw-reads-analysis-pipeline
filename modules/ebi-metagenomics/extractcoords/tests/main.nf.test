
nextflow_process {

    name "Test Process EXTRACTCOORDS"
    script "../main.nf"
    process "EXTRACTCOORDS"

    tag "modules"
    tag "modules_nfcore"
    tag "extractcoords"

    test("extractcoords - should take a fasta file and an easel coordinates file and extract two subunit fasta files for the SSU and LSU, and a concatenated SSU and LSU coordinates file") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test.fasta", checkIfExists: true)
                    ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test.matched_seqs_with_coords.txt", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("extractcoords - stub") {

        options "-stub"

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test.fasta", checkIfExists: true)
                    ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test.matched_seqs_with_coords", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
