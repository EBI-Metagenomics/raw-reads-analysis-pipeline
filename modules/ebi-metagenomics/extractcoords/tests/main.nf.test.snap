{
    "extractcoords - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,bb41e1add63c9b89d054972d8f663caf"
                ],
                "concat_ssu_lsu_coords": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "lsu_fasta": [
                    
                ],
                "ssu_fasta": [
                    
                ],
                "versions": [
                    "versions.yml:md5,bb41e1add63c9b89d054972d8f663caf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-05-09T14:32:05.714677142"
    },
    "extractcoords - should take a fasta file and an easel coordinates file and extract two subunit fasta files for the SSU and LSU, and a concatenated SSU and LSU coordinates file": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_SSU.fasta:md5,fb23f13526672a300845b767dbc5fd44"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords.txt:md5,401222d263116bb145c1a67e6acc4709"
                    ]
                ],
                "3": [
                    "versions.yml:md5,bb41e1add63c9b89d054972d8f663caf"
                ],
                "concat_ssu_lsu_coords": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords.txt:md5,401222d263116bb145c1a67e6acc4709"
                    ]
                ],
                "lsu_fasta": [
                    
                ],
                "ssu_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_SSU.fasta:md5,fb23f13526672a300845b767dbc5fd44"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bb41e1add63c9b89d054972d8f663caf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-05-09T14:31:38.962282816"
    }
}